### ISSUE TITLE
ONNX skill-selector model inference failure

### ISSUE THREAD

**Issue by chalie-ai** [chalie-ai/chalie#592]
## Problem
The skill-selector ONNX model loads successfully but fails during inference with an array indexing error.

## Error
```
Multi-label inference failed for skill-selector: too many indices for array: array is 2-dimensional, but 3 were indexed
```

## Details
- Model: skill-selector (one of three expected ONNX models)
- Phase: Runtime inference (not load)
- Root cause: Array shape mismatch — the inference code attempts to index a 2D array as if it were 3D
- Scenario context: Detected in nightly scenario test "ONNX models loaded at boot without errors"

## Likely causes
1. Model output shape has changed or is incompatible with current inference code
2. Input shape passed to inference is incorrect
3. Model file corruption or version mismatch
4. Bug in multi-label inference output handling

## Impact
- ONNX models cannot be used for inference at boot
- skill-selector functionality is broken

## Detection
Found in Step 2 of scenario test execution: WARNING from ONNX inference service

**Comment by chalie-ai** [comment]
Root cause: skill-selector classifier_meta.json was missing `"pruned": true`. The model outputs 2D tensors (pruned classification head) but the inference code defaulted to the unpruned 3D path (`logits[0, seq_len, :]`), causing the array indexing crash.

Fixed locally and in the GitHub release (chalie-ai/models v0.3.0 — replaced skill_selector.json asset with corrected metadata).
